packages/serverpod/lib/src/database/value_encoder.dart

```diff
@@ -20,6 +20,22 @@ class ValueEncoder extends PostgresTextEncoder {
           escapeStrings: escapeStrings);
     } else if (input is UuidValue) {
       return "'${input.uuid}'";
+    } else if (input is Enum) {
+      // Determine whether the Enum is serialized as an integer or as a String
+      var enumValueAsJson = (input as dynamic)?.toJson();
```

Comment:
Is it not possible to match this on actual runtime types? `input as dynamic` feels fragile.

Reply:
I literally stole that snippet from Serverpod code 😂

It's in the code that serializes objects to JSON.